Hazelcast.Test/Hazelcast.Client.Test/ClientMapTest.cs

```diff
@@ -592,6 +593,34 @@ public void TestListener()
         }
 
         [Test]
+        public void TestListener_SingleEventListeners()
+        {
+            var listener = new ListenerImpl<object, object>();
+            var reg1 = map.AddEntryListener(listener, false);
+
+            map.Put("key1", "value1");
+            map.Put("key1", "value2");
+            map.Remove("key1");
+            map.Put("key1", "value2");
+            map.Clear();
+            map.Put("key1", "value2");
+            map.EvictAll();
+            map.Put("key2", "value2");
+            map.Evict("key2");
+            map.Put("key3", "value2", 1l, TimeUnit.Seconds);
+
+            Assert.IsTrue(listener.GetLatch(EntryEventType.Added).WaitOne(TimeSpan.FromSeconds(10)));
```

Comment:
You may want to put this line right after map.Put("key1", "value1"); just to make sure that the event listener update did not mistakenly update the wrong latch but at the end since all latches are set, even if it is updated wrong (i.e. wrong callback method), the test will pass.

Reply:
make sense, I'll fix it